scip-java/src/main/scala/com/sourcegraph/scip_java/buildtools/BazelBuildTool.scala

```diff
@@ -13,15 +26,216 @@ class BazelBuildTool(index: IndexCommand) extends BuildTool("Bazel", index) {
   }
 
   override def generateScip(): Int = {
-    index
-      .app
-      .reporter
-      .error(
-        "The `scip-java index` command does not work with Bazel builds because Bazel " +
-          "requires custom configuration to be added to WORKSPACE and BUILD files. " +
-          "The scip-java repository contains an example WORKSPACE file to demonstrate " +
-          "how to use scip-java with Bazel https://github.com/sourcegraph/scip-java/blob/main/examples/bazel-example/WORKSPACE"
-      )
-    1
+    this.generateAspectFile() match {
+      case None =>
+        1
+      case Some(aspectLabel) =>
+        val buildCommand = ListBuffer.empty[String]
+        buildCommand += "bazel"
+        buildCommand += "build"
+        buildCommand += "--noshow_progress"
+
+        // The local strategy is required for now because we write SemanticDB and SCIP files
+        // to the provided targetroot directory.
+        buildCommand += "--spawn_strategy=local"
+
+        val targetSpecs =
+          if (index.buildCommand.isEmpty)
+            List("//...")
+          else
+            index.buildCommand
+        buildCommand ++= targetSpecs
+
+        buildCommand += "--aspects"
+        buildCommand += s"$aspectLabel%scip_java_aspect"
+        buildCommand += "--output_groups=scip"
+        buildCommand += s"--define=sourceroot=${index.workingDirectory}"
+
+        val javaHome = index
+          .app
+          .env
+          .environmentVariables
+          .getOrElse("JAVA_HOME", "")
+        if (javaHome.isEmpty) {
+          index
+            .app
+            .error(
+              "environment variable JAVA_HOME is not set. " +
+                "To fix this problem run `export JAVA_HOME=/path/to/java` and run scip-java index again."
+            )
+          return 1
```

Comment:
This entire new logic feels very much like it's written in Java, with Scala syntax.

Would be better to 

1. Remove the builder, replacing it with just a regular list, as I think all the places where you append to it are trivially replaced with 
2. May be using a beefed up assert method that throws (and logs the error through `index.app.error`)? There's several early exists in this function that IMO doesn't have to be there